File: code/java/dsl/01.java
val order = Sandwich()
    .type("toastie")
    .bread("white")
    .fillings(List.of("Cheese", "Ham"))
    .dressings(List.of("Pepper", "Worcestershire sauce"))
    .side("Crisps")
    .makeMeASandwich()

System.out.println(ooSandwich.receipt())

// https://www.kotlindays.com/2019/12/02/days-of-kotlin-a-dsl-for-everyone/index.html

File: code/kotlin/dsl/01.kt
val dslSandwich = sandwich {
    with type "toasted"
    bread = "baguette"

    filling("cheese")
    filling("ham")
    filling("tomato")

    dressings {
        +"Basil"
        +"Pepper"
    }

    sideOrders {
        side("French Fries")
    }

}

println(dslSandwich.receipt())

// https://www.kotlindays.com/2019/12/02/days-of-kotlin-a-dsl-for-everyone/index.html